icu4c/source/test/intltest/usettest.cpp

```diff
@@ -3984,3 +3985,31 @@ void UnicodeSetTest::TestDeepPattern() {
     assertTrue("[a[a[a...1000s...]]] -> error", errorCode.isFailure());
     errorCode.reset();
 }
+
+void UnicodeSetTest::TestEmptyString() {
+    IcuTestErrorCode errorCode(*this, "TestEmptyString");
+    // Starting with ICU 69, the empty string is allowed in UnicodeSet. ICU-13702
+    UnicodeSet set(u"[{}]", errorCode);
+    if (!assertSuccess("set from pattern with {}", errorCode)) { return; }
+    assertTrue("set from pattern with {}", set.contains(u""));
+    assertFalse("remove empty string", set.remove(u"").contains(u""));
+    assertTrue("add empty string", set.add(u"").contains(u""));
+    // missing API -- assertTrue("retain empty string", set.retain(u"").contains(u""));
+    assertFalse("complement-remove empty string", set.complement(u"").contains(u""));
+    assertTrue("complement-add empty string", set.complement(u"").contains(u""));
+    // no C++ equivalent for Java strings() -- assertTrue("strings()", set.strings().contains(u""));
+    UnicodeSetIterator sit(set);
+    assertTrue("set iterator.next()", sit.next());
+    assertTrue("set iterator has empty string", sit.isString() && sit.getString().isEmpty());
+    // The empty string is ignored in matching.
```

Comment:
I'm confused by this test because you add and remove the empty string too much. It's hard to convince myself when reading this test that the set actually contains the empty string at this point. It would be more clear if you had a line explicitly resetting the set before each chunk of tests.

Reply:
added comments and empty lines